### PR TITLE
Explain SSH agent proxy functionality

### PR DESCRIPTION
**Pull request by GMTA**
See https://github.com/docksal/docksal/issues/1239

### CODE REVIEW

File: docs/content/core/system-ssh-agent.md [hunk 1]
```diff
@@ -7,28 +7,16 @@ aliases:
...
 
 The ssh-agent service allows adding multiple keys (including the ones protected with a passphrase) to a single pool of 
-ssh keys, which can be shared across multiple projects.
+SSH keys, which can be shared across multiple projects. It runs an SSH agent process whose Unix socket is mounted into
+your project's cli container by default.
 
-The default ssh keys (`~/.ssh/id_rsa`, `~/.ssh/id_dsa`, `~/.ssh/id_ecdsa`) are loaded into the agent automatically.  
-On macOS and Windows this happens when the Docksal VM is (re)started, on Linux - whenever `fin project start` is used.
+The default SSH keys (`~/.ssh/id_rsa`, `~/.ssh/id_dsa`, `~/.ssh/id_ecdsa`, `~/.ssh/id_ed25519`) are loaded into the
+agent automatically. On macOS and Windows this happens when the Docksal VM is (re)started, on Linux - whenever
```
Comment (shelane) [inline docs/content/core/system-ssh-agent.md]: Add comma after “and Windows” and change dash after Linux to comma.

File: docs/content/core/system-ssh-agent.md [hunk 2]
```diff
@@ -73,7 +63,27 @@ SECRET_SSH_KEY_ACQUIA='acquia_key'
...
+  * By setting the environment variable `DOCKSAL_SSH_AGENT_USE_HOST` to `1`. This will explictly enable proxy mode. Set
+    it by running `fin config set --global DOCKSAL_SSH_AGENT_USE_HOST=1` followed by `fin system reset ssh-agent`.
+
+On Linux, ssh-agent is able to directly mount your host SSH agent's Unix socket into the container.
+
+On Windows and macOS, the `socat` utility is used to convert your local Unix socket to a TCP-endpoint and back again.
+To do so, a local port configured by the `DOCKSAL_SSH_PROXY_PORT` environment variable (defaults to `30001`) is opened.
+
+**Caution:** if other users have local access to your system, they could connect to this port and use your keys. The
+port is only opened on an internal network interface and is not accessible from e.g. your LAN. Use with care!
```
Comment (shelane) [inline docs/content/core/system-ssh-agent.md]: Are you using “e.g.” to mean “for example,” or is it a definitive statement? It’s placement here is awkward.

File: docs/content/core/system-ssh-agent.md (hunk 1 above)
Comment (GMTA) [reply docs/content/core/system-ssh-agent.md]: I've updated this sentence

File: docs/content/core/system-ssh-agent.md (hunk 2 above)
Comment (GMTA) [reply docs/content/core/system-ssh-agent.md]: It's meant to be read as "for example". Do you have a suggestion for a better sentence structure? My goal is to differentiate between local and external network access.

Comment (shelane) [reply docs/content/core/system-ssh-agent.md]: I think just removing “e.g.” here would make it read better.

Comment (GMTA) [reply docs/content/core/system-ssh-agent.md]: Updated the statement.

**Comment by achekulaev** [comment]
Thanks @GMTA and @shelane

**Pull request merged**